docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,265 @@
+
+### Options & Command-line interface
+
+Our stable and supported API surface includes the k6 options set (as exposed to users via the various configuration mediums k6 supports, options object, config file, environment variables, etc.).
+
+The k6 command-line interface (CLI), including commands, subcommands, and flags, is also part of the stable and supported API. These commands are essential for initiating and managing test executions locally and in the cloud.
+
+### JavaScript interpreter
+
+With k6 scripts written in JavaScript and the environment in which they are executed supporting a specific set of JavaScript syntax and ECMAScript features, the k6 JavaScript interpreter is considered part of the k6 stable and supported API surface.
+
+This ensures that users can write and execute scripts with predictable behavior and compatibility over the lifespan of their major version of k6.
+### Modules
+
+Furthermore, the *“core”* modules exposed directly to a script by k6 *(e.g.,* `k6`, `k6/http`, `k6/execution`) are another critical aspect of the k6 experience and of the interface between k6 and its users and is as such part of the stable and supported API surface.
+It includes any class, properties, methods, function, or relative signature that is part of the module and is publicly exposed and documented.
+
+Here, the full list of the `core` modules: `k6`, `k6/browser`, `k6/crypto`, `k6/crypto/x509`, `k6/data`, `k6/encoding`, `k6/execution`,  `k6/http`, `k6/metrics`, `k6/net/grpc`, `k6/timers`, `k6/ws`.
+
+Experimental modules recognizable from their `k6/experimental` prefixed import path are explicitly excluded from the stable and supported API surface.
+### Extensions
+
+Our stable and supported API surface also includes the extensions API, which empowers internal and external developers to build custom modules and outputs as part of the k6 codebase or as external extensions.
+
+This includes the [go.k6.io/k6/js/modules](https://pkg.go.dev/go.k6.io/k6@v0.57.0/js/modules) Go package that internal and external developers have available to build module extensions, as well as the [go.k6.io/k6/output](https://pkg.go.dev/go.k6.io/k6@v0.57.0/output) that can be used to build custom outputs for k6.
+
+It also includes public k6 Go packages that are exposed to developers to facilitate the development of these extensions (e.g., the current  [go.k6.io/k6/lib.State](https://pkg.go.dev/go.k6.io/k6@v0.57.0/lib#State) package).
+### Output
+
+We guarantee the stability and support of metrics, logs, traces, and profiling outputs that we fully control in an end-to-end flow. This includes the JSON, CSV, JSON summary exports, cloud metrics, logs and traces outputs, and OpenTelemetry formats while excluding outputs dependent on third-party platforms and protocols.
```

Comment:
If I recall correctly, the new JSON format for the summary output will not be introduced in `v1.0.0`. Therefore, according to this, it won't be introduced until at least `v2.0.0` or we simply consider [the existing `JSON` format ](https://grafana.com/docs/k6/latest/results-output/end-of-test/#summary-export-to-a-json-file)deprecated and therefore changing it is no a "breaking"

Reply:
Yeah, good catch. This was added mostly because in our intentions but we haven't accomplished it. I am dropping it.

@joanlopez @oleiade do we introduce any guarantee for the new End-of-Test summary? I guess we should.

Reply:
@codebien we have had a couple discussions about this in the past. My stance is the end-of-test summary (both human readable and machine readable) is explicitly excluded from our stability policies and guarantees in v1.0.0. I expect v1.0.0 to explicitly mark the existing json end of test summary as deprecated.

As we go through the development cycle towards v2.0.0, we may ship a new, supported this time, machine readable end of test summary. At this point, then and only then, do I suggest we add the machine readable summary to this document, with the option to extend it to the new summary/report code and any potential summary extension API we might come up with, at that point in time.

Any strong objections?